File: Easy/796.java
class Solution {
    public boolean rotateString(String s, String goal) {
        // check if the size of both strings are equal or not if not then its not a variation of string s.
        if(s.length() != goal.length()) return false;
        
        // creating a new string with all possible combinations. 
        String check = s+s;

        //check if it contains goal or not and returing it.
        return check.contains(goal);
    }
}

File: Easy/796.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
        int n = s.length();
        bool ans = false;
        if(n != goal.length()) return false;
        for(int i=0;i<n;i++){
        int temp = s[n-1];
        for(int j=n-1;j>=0;j--){    
            if(j!=0){
                s[j] = s[j-1];
            }else{
                s[j] = temp;    
            }
            
            cout<<"Iteration of j "<<j<<" : "<<s<<endl;
        }
        if(s == goal){
            ans = true;
            break;
        }
        cout<<"Iteration "<<i+1<<" : "<<s<<endl;
        }
        return ans;
    }
};